src/Exceptions/ImpersonationTokenCouldNotBeCreatedWithNonStatefulGuard.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Stancl\Tenancy\Exceptions;
+
+use Exception;
+
+class ImpersonationTokenCouldNotBeCreatedWithNonStatefulGuard extends Exception
```

Comment:
I'd simplify the name, it's too long now. Also, all exceptions should be suffixed by `Exception`. That'd make the name even longer.

Perhaps the best way to simplify this is to remove the context/intended action, and only keep the actual error. Meaning: don't make the exception about creating impersonation tokens, instead make it simply say that a stateful guard is required.

Reply:
https://github.com/archtechx/tenancy/pull/935/commits/f1e93a0ca0d1dfaf9f0c86a41da9a9462c32cf81 I updated the message too